lms/djangoapps/courseware/access.py

```diff
@@ -370,17 +396,18 @@ def _can_load_course_overview(user, course_overview):
         The user doesn't have to be enrolled in the course in order to have load
         load access.
     """
-    return (
-        not course_overview.visible_to_staff_only
+    return _has_staff_access_to_descriptor(user, course_overview, course_overview.id) or \
+        (
+        _is_visible_to_staff_only(course_overview)
```

Comment:
`course_overview.visible_to_staff_only` is a boolean, `_is_visible_to_staff_only(course_overview)` uses that boolean to return the correct corresponding AccessResponse object